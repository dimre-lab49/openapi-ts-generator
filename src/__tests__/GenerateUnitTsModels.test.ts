import { existsSync, readdirSync, readFileSync } from 'fs';
import { generateTsModels } from '../index';

beforeAll(async () => {
  await generateTsModels(
    'https://d-unit-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
    './output_unt/',
  );
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_unt');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an output directory', async done => {
  try {
    const files = await readdirSync('./output_unt');
    expect(files.length).toBeGreaterThan(0);
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('Generate TS Models should generate an index.ts barrel file', async done => {
  try {
    const fileExist = await existsSync('./output_unt/index.ts');
    expect(fileExist).toBeTruthy();
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('IUnit should be correct', async done => {
  try {
    const file = await readFileSync('./output_unt/unit.model.ts');
    expect(file.toString()).toMatchSnapshot();
    done();
  } catch (err) {
    done.fail(err);
  }
});

test('UnitFormGroupFac should be correct', async done => {
  try {
    const file = await readFileSync('./output_unt/unit.form-group-fac.ts');
    expect(file.toString()).toMatchSnapshot();
    done();
  } catch (err) {
    done.fail(err);
  }
});
