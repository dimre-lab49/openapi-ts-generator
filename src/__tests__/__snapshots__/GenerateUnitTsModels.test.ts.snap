// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IUnit should be correct 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-model-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
 // tslint:disable: array-type
 
import { IBuilding } from './building.model';
import { ISite } from './site.model';
import { IUnitTag } from './unit-tag.model';
import { IUnitType } from './unit-type.model';

export interface IUnit {
    building?: IBuilding;
    buildingId?: string;
    createdBy?: string;
    createdOnUtc?: Date;
    id?: string;
    isEnabled?: boolean;
    number?: string;
    site?: ISite;
    siteId?: string;
    unitTags?: Array<IUnitTag>;
    unitTypeId?: string;
    unitTypeName?: string;
    unitTypes?: IUnitType;
    updatedBy?: string;
    updatedOnUtc?: Date;
}

export enum UnitProperties {
    BUILDING = 'building',
    BUILDING_ID = 'buildingId',
    CREATED_BY = 'createdBy',
    CREATED_ON_UTC = 'createdOnUtc',
    ID = 'id',
    IS_ENABLED = 'isEnabled',
    NUMBER = 'number',
    SITE = 'site',
    SITE_ID = 'siteId',
    UNIT_TAGS = 'unitTags',
    UNIT_TYPE_ID = 'unitTypeId',
    UNIT_TYPE_NAME = 'unitTypeName',
    UNIT_TYPES = 'unitTypes',
    UPDATED_BY = 'updatedBy',
    UPDATED_ON_UTC = 'updatedOnUtc',
}
"
`;
